packages/datadog-plugin-cucumber/src/index.js

```diff
@@ -0,0 +1,95 @@
+const { relative } = require('path')
+
+const { SAMPLING_RULE_DECISION } = require('../../dd-trace/src/constants')
+
+const {
+  TEST_TYPE,
+  TEST_NAME,
+  TEST_SUITE,
+  TEST_STATUS,
+  getTestEnvironmentMetadata
+} = require('../../dd-trace/src/plugins/util/test')
+
+function setStatusFromResult (span, result) {
+  if (result.status === 1) {
+    span.setTag(TEST_STATUS, 'pass')
+  } else if (result.status === 2) {
+    span.setTag(TEST_STATUS, 'skip')
+    span.setTag('error.msg', 'skipped')
+  } else if (result.status === 4) {
+    span.setTag(TEST_STATUS, 'skip')
+    span.setTag('error.msg', 'not implemented')
+  } else {
+    span.setTag(TEST_STATUS, 'fail')
+    span.setTag('error.msg', result.message)
+  }
+}
+
+function createWrapRun (tracer, testEnvironmentMetadata, sourceRoot) {
+  return function wrapRun (run) {
+    return async function handleRun () {
+      const testName = this.pickle.name
+      const testSuite = relative(sourceRoot, this.pickle.uri)
+
+      const commonSpanTags = {
+        [TEST_TYPE]: 'test',
+        [TEST_NAME]: testName,
+        [TEST_SUITE]: testSuite,
+        [SAMPLING_RULE_DECISION]: 1,
+        ...testEnvironmentMetadata
+      }
+
+      const result = await tracer.trace(
+        'cucumber.test',
+        {
+          type: 'test',
+          resource: testName,
+          tags: commonSpanTags
+        },
+        async (span) => {
+          const runResult = await run.apply(this, arguments)
```

Comment:
Generally speaking, we try to avoid async/await in the tracer because it creates 3 promises for every actual promise needed. Given the issues with `async_hooks`, it's best to not make it worse with additional promises that are avoidable. If all that is needed here is to add a tag after the promise has resolved, this can be replaced by this snippet:

```js
const promise = run.apply(this, arguments)

promise.then(() => {
  setStatusFromResult(span, this.getWorstStepResult())
})

return promise
```

That will also remove the need for all the other async/awaits in the function.